cursive/examples/filters.rs

```diff
@@ -0,0 +1,135 @@
+use cursive::view::Resizable;
+use cursive::views::{Dialog, LinearLayout, RadioGroup, TextView};
+
+#[derive(Clone, Copy, Debug)]
+enum Color {
+    Red,
+    Blue,
+}
+
+#[derive(Clone, Copy, Debug)]
+enum Size {
+    Large,
+    Small,
+}
+
+#[derive(Default)]
+struct Filters {
+    color: Option<Color>,
+    size: Option<Size>,
+}
+
+// This is an examples of a Settings/Configuration "pop-up" where we can
+// modify user_data to be used elsewhere in our application.
+
+fn main() {
+    let mut siv = cursive::default();
+    siv.set_user_data(Filters::default());
+    // Leave the menu visible
+    siv.set_autohide_menu(false);
+    siv.menubar()
+        .add_leaf("Filters", |s| {
+            // Hide menu while "pop-up" settings are visible
+            s.set_autohide_menu(true);
```

Comment:
I think this is more up to personal preference, and could be removed from the example to keep it more focused.

Reply:
I left this in because otherwise we could keep opening "Filter" windows on top of each other. I agree it has nothing to do with the example; it just looks nicer. Let me know if you still want me to remove it.